experiment/coverage/apicoverage_test.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"regexp"
+	"testing"
+)
+
+//NOTE: This method is for avoiding flake tests instead of using reflect.DeepEqual()
+func equalAPIArray(a, b apiArray) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if a == nil || b == nil {
+		return false
+	}
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		found := false
+		for j := range b {
+			if i == j {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+func TestParseOpenAPI(t *testing.T) {
+	testCases := []struct {
+		Rawdata  []byte
+		Expected apiArray
+	}{
+		{
+			Rawdata: []byte(`{"paths": {"/resources": {
+				"get": {"description": "get available resources"}}}}`),
+			Expected: apiArray{
+				apiData{Method: "GET", URL: "/resources"},
+			},
+		},
+		{
+			Rawdata: []byte(`{"paths": {"/resources": {
+				"get": {"description": "get available resources"},
+				"post": {"description": "create resource"}}}}`),
+			Expected: apiArray{
+				apiData{Method: "GET", URL: "/resources"},
+				apiData{Method: "POST", URL: "/resources"},
+			},
+		},
+		{
+			Rawdata: []byte(`{"paths": {
+			"/resources": {
+				"get": {"description": "get available resources"},
+				"post": {"description": "create resource"}},
+			"/foo": {
+				"get": {"description": "get avaiable foo"},
+				"post": {"description": "create foo"},
+				"parameters": [{"type": "string", "description": "This should be ignored", "name": "bar", "in": "query"}]}}}`),
+			Expected: apiArray{
+				apiData{Method: "GET", URL: "/resources"},
+				apiData{Method: "POST", URL: "/resources"},
+				apiData{Method: "GET", URL: "/foo"},
+				apiData{Method: "POST", URL: "/foo"},
+			},
+		},
+	}
+	for _, test := range testCases {
+		res := parseOpenAPI(test.Rawdata)
+		if !equalAPIArray(res, test.Expected) {
+			t.Errorf("OpenAPI did not match expected for test")
+			t.Errorf("%#v", res)
+			t.Errorf("%#v", test.Expected)
+		}
+	}
+}
+
+//func getTestedAPIsByLevel(reg *regexp.Regexp, apisOpenapi, apisTested apiArray) (apiArray, apiArray) {
+func TestGetTestedAPIsByLevel(t *testing.T) {
```

Comment:
I think an important thing to test here is `parseAPILog()`, namely that the following:
```
round_trippers.go:444] GET https://foo/bar?other
round_trippers.go:444] POST https://this/that?thing
```

gets translated into:
```
apiArray{
  { Method: "GET", URL: "https://foo/bar" },
  { Method: "POST", URL: "https://this/that" },
}
```

Reply:
That is a nice point I wanted to do that before.
Current parseAPILog seems difficult to be applied to gomock for os.Open() and bufio.NewReaderSize() because the corresponding interfaces are not implemented, so I need more time for implementing the unit test.
Hopefully I will be able to do it soon.